### ISSUE TITLE
[Question] Anyone know of an EAP library?

### ISSUE THREAD

**Issue by ByronAP** [vforteli/Flexinets.Radius.Core#9]
Anyone know of an EAP library that can be used with this?

**Comment by vforteli** [comment]
I dont think there should be any problems implementing eap of some sort on top of this but i havent had the need or time. Then there is also tje question of which eap flavor?

**Comment by ByronAP** [comment]
by flavor I assume you mean which auth scheme?

**Comment by vforteli** [comment]
Yes exactly

**Comment by ByronAP** [comment]
I think only a few really would need to be added since most only use a select few like mschapv2, I think md5 is kinda depreciated, etc...

**Comment by vforteli** [comment]
Mschapv2 and eap-tls i guess.
I actually got a bit inspired so if i have the time ill probably add eap message parsing and creation. Then it would also need to handle fragmented packets. 
Then it would of course still need to do something sensible with the eap messages

**Comment by ByronAP** [comment]
Invoke a plugin :-) like everything else does. I have a guy working on adding it but he may have questions so I refereed him to this thread.

**Comment by neals-sudo** [comment]
Hey @vforteli , I am the guy @ByronAP referring to,  The biggest issue I had came across is failure to parse all attributes by RadiusPacketParser library even though it has the same secret key, and with a bit of research it seems, it's using PEAP and EAP-TLS in top to be able to access the other attributes like User-Password, those other attributes are crucial to to implement the Mschapv2 handshake. 

How would you advise me to do this in an efficient way?

**Comment by vforteli** [comment]
Thats the thing, if its EAP the interesting attribute is eap-message. There can be several in one packet and then they have to be concatenated and then parsed. This should be doable with the library with a custom packet handler. Also if its EAP it shouldnt have a user-password at all

**Comment by ByronAP** [comment]
Yeah it's just a lot of reading for us both as we try to figure out the whole flow and back and encryption/decryption properly. It is very cryptic when looking at the actual eap message bytes which I guess is the point and the docs are very scattered and seem to have t o be pieced together to make a working flow from what I have seen.
